main.js

```diff
@@ -1,82 +1,158 @@
-        } else {
-            document.getElementById('calc').value = number;
-        }
+    let currentNumber = document.getElementById('calc').value;
+    //debugger;
+    if ( currentNumber.indexOf('.') !== -1 ) {
+        currentNumber = currentNumber + number;
+        numInputs[numInputs.length - 1] = currentNumber;
+    } else if ( currentNumber === '0' ) {
+        currentNumber = number;
+        numInputs.push(currentNumber);
     } else {
-        document.getElementById('calc').value = document.getElementById('calc').value + number;
+        currentNumber = currentNumber + number;
+        numInputs[numInputs.length - 1] = currentNumber;
     }
+    document.getElementById('calc').value = currentNumber;
+    createHist();
+    clearOp();
 }
 
-function allClearNum() {
-    // clears input to 0 and clears all variables
-    document.getElementById('calc').value = 0;
-    val1 = undefined;
-    val2 = undefined;
-    op = undefined;
+function inputOp(op) {
+    let currentOp = document.getElementById('op').value;
+
+    if ( currentOp === '' ) {
+        opInputs.push(op);
+    } else {
+        opInputs[opInputs.length - 1] = op;
+    } 
+    document.getElementById('op').value = op;
+    clearNum();
+    createHist();
 }
 
 function makeNeg() {
-    if (document.getElementById('calc').value == 0 || isNaN(document.getElementById('calc').value)) {
+    let currentNumber = document.getElementById('calc').value;
+
+    if ( currentNumber === 0 || isNaN( currentNumber )) {
         // allows a leading 0 with decimal to happen
-        document.getElementById('calc').value = 0;
-    } else {
-        document.getElementById('calc').value = '-' + document.getElementById('calc').value;
-    }
+        return;
+    } 
+    currentNumber = Number( currentNumber ) * -1;
+    numInputs[numInputs.length - 1] = currentNumber;
+    document.getElementById('calc').value = currentNumber;
 }
 
-function clearNum() {
-    document.getElementById('calc').value = 0;
+//create history to display
+function createHist() {
+    let hist = '';
+    for ( i = 0; i < numInputs.length; i++) {
+        hist += numInputs[i];
+        if ( opInputs[i] !== undefined ) {
+            hist += opInputs[i];
+        }
+    }
+    document.getElementById('hist').innerHTML = hist;
 }
 
-function decimal() {
-    if (document.getElementById('calc').value == 0 || isNaN(document.getElementById('calc').value)) {
-        // allows a leading 0 with decimal to happen
-        document.getElementById('calc').value = 0 + '.';
-    } else {
-        if (document.getElementById('calc').value.includes('.')) {
-            // stops user from entering multiple decimals
-            document.getElementById('calc').value=document.getElementById('calc').value;
+
+//Calculate functions
+function calculate() {
+    const operators = ['*', '/', '+', '-'];
+    let opIndex = 0;
+    while ( opInputs.length > 0 ) {
```

Comment:
You should change this to be be `opInputs.length > 0 && opIndex < operators.length` to avoid running into a possible infinite loop like we had previously. that will ensure we break out of the loop if we do not find any operators in the calculation.